collect/process.go

```diff
@@ -42,40 +43,48 @@ func ProcessMetrics(prometheusAPI v1.API, metric string, metricObjects []metrics
 	}
 
 	for _, obj := range metricObjects {
-		dumpFile, err := createDumpFilePath(dataDir, metric, obj.Filename, timePeriod)
-		if err != nil {
-			return fmt.Errorf("error creating dump file path: %w", err)
-		}
-		defer dumpFile.Close()
+		if err := func() error {
```

Comment:
The function `ProcessMetrics` includes some complex metric handling logic. On one hand, it is a good practice to implement a deep method and hide the complexity from the callers. On the other hand, because the method is deep, we need to carefully unit-test it.

Currently, it's tricky to unit-test, because it writes to a file. Could you break this method into two parts, one is to prepare the format and args, the other is to write to the file. That way, we can unit-test the first part by preparing mock metrics and verifying the outputted format and args.

Reply:
Additionally, when a method is deep, we need to make sure the descriptions are concise. Please add descriptions for each of the input parameters and more details on how the data gets written to what file. e.g. when the caller tries to invoke this method, looking at the method definition and description, it has no idea timePeriod is second or minute.